src/AI.java

```diff
@@ -151,55 +149,57 @@ else if (curr.color < 3 && curr.rowSize != 3) {
 				if (player == 2) { //AI - find largest value
 					//check if this position already gives a win
 					if (b.checkWin(x, i, 2)) {
-						ans.add(Integer.MIN_VALUE);
+						ans.put(i, Integer.MIN_VALUE);
 						if (abPruning)
 							beta = Integer.MIN_VALUE;
 						continue;
 					}
 
-					value.addAll(minMax(b, depth - 1, 1, alpha, beta));
+					value = minMax(b, depth - 1, 1, alpha, beta);
 
-					int max = Integer.MIN_VALUE;
-					for (int j : value) {
-						max = Math.max(max, j);
+					int max = Integer.MIN_VALUE, index = -1;
+					for (Map.Entry<Integer, Integer> j : value.entrySet()) {
```

Comment:
If possible, maybe try to use TreeMap's ability to sort things to grab the largest key. That probably needs a lot of refactoring, though.

Reply:
[Here's how to sort by value](https://stackoverflow.com/questions/2864840/treemap-sort-by-value). Kind of difficult to implement